File: Python/Prob10.py
#stackoverflows, the haskell one works

primes = []

def isPrime(num, d):
	if d*d > num: return True
	elif num%d==0: return False
	else: return isPrime(num, d+1)

for i in xrange(2000000):
	if isPrime(i, 2):
		primes.append(i)

print sum(primes)

File: Haskell/prob10.hs
--slow, but got there eventually.

isPrime n = go 2
	where
		go d
			| d*d > n = True
			| n `mod` d == 0 = False
			| otherwise = go (d+1)

primes = 2:filter isPrime [3,5..]



main = print $ sum $ takeWhile (<2000000) primes